utils/filesystem/files_test.go

```diff
@@ -2030,3 +2030,261 @@ func TestConvertToOSFile(t *testing.T) {
 	require.NoError(t, osFile.Close())
 	require.Error(t, rFile.Close(), "file must already be closed")
 }
+
+func createTestFileStructure(vfs afero.Fs, rootDir string) error {
+	dirs := []string{
+		filepath.Join(rootDir, "dir2", "dir3", "dir4"),
+		filepath.Join(rootDir, "dir2", "dir5"),
+	}
+	files := []string{
+		filepath.Join(rootDir, "dir2", "dir3", "dir4", "test1.txt"),
+		filepath.Join(rootDir, "dir2", "dir3", "dir4", "test2.o"),
+		filepath.Join(rootDir, "dir2", "dir3", "dir4", "test3.h"),
+		filepath.Join(rootDir, "dir2", "dir5", "test4.txt"),
+		filepath.Join(rootDir, "dir2", "dir5", "test5.jar"),
+	}
+
+	for _, dir := range dirs {
+		if err := vfs.MkdirAll(dir, 0755); err != nil {
+			return err
+		}
+	}
+	for _, file := range files {
+		err := afero.WriteFile(vfs, file, []byte("test content"), 0644)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func TestVFS_LsRecursive(t *testing.T) {
+	fs := globalFileSystem
+
+	vfs := afero.NewOsFs()
+
+	rootDir, err := afero.TempDir(vfs, "", "testdata")
+	if err != nil {
+		t.Fatalf("Failed to create temp directory: %v", err)
+	}
+	defer func() { _ = vfs.RemoveAll(rootDir) }()
```

Comment:
you can just do `rootDir := t.TempDir()` and not worry about the cleanup or errors.